api/src/main/java/com/github/streamshub/console/api/ClientFactory.java

```diff
@@ -552,26 +566,31 @@ public void disposeKafkaContext(@Disposes KafkaContext context, Map<String, Kafk
 
     @Produces
     @RequestScoped
-    public Supplier<Consumer<byte[], byte[]>> consumerSupplier(ConsoleConfig consoleConfig, KafkaContext context) {
+    public Consumer<RecordData, RecordData> consumerSupplier(ConsoleConfig consoleConfig, KafkaContext context) {
         var configs = maybeAuthenticate(context, Consumer.class);
-        Consumer<byte[], byte[]> client = new KafkaConsumer<>(configs); // NOSONAR / closed in consumerDisposer
-        return () -> client;
+
+        return new KafkaConsumer<>(
+                configs,
+                context.schemaRegistryContext().keyDeserializer(),
+                context.schemaRegistryContext().valueDeserializer());
     }
 
-    public void consumerDisposer(@Disposes Supplier<Consumer<byte[], byte[]>> consumer) {
-        consumer.get().close();
+    public void disposeConsumerSupplier(@Disposes Consumer<RecordData, RecordData> consumer) {
```

Comment:
Why the word Supplier in this method name?

Reply:
I'll remove it. The name is a relic of the old type being `Supplier<Consumer<x, x>>`